packages/sensors/example/lib/snake.dart

```diff
@@ -0,0 +1,114 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:flutter/material.dart';
+import 'package:sensors/sensors.dart';
+
+class Snake extends StatefulWidget {
+  final int rows;
+  final int columns;
+  final double offsetSize;
+  Snake({this.rows = 20, this.columns = 20, this.offsetSize = 10.0});
+
+  @override
+  State<StatefulWidget> createState() =>
+      new SnakeState(rows, columns, offsetSize);
+}
+
+class SnakeBoardPainter extends CustomPainter {
+  GameState state;
+  double offsetSize;
+
+  SnakeBoardPainter(this.state, this.offsetSize);
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final blackLine = new Paint()..color = Colors.black;
+    final blackFilled = new Paint()
+      ..color = Colors.black
+      ..style = PaintingStyle.fill;
+    canvas.drawRect(
+        new Rect.fromPoints(Offset.zero, size.bottomLeft(Offset.zero)),
+        blackLine);
+    for (math.Point<int> p in state.body) {
+      final a = new Offset(offsetSize * p.x, offsetSize * p.y);
+      final b = new Offset(offsetSize * (p.x + 1), offsetSize * (p.y + 1));
+
+      canvas.drawRect(new Rect.fromPoints(a, b), blackFilled);
+    }
+  }
+
+  @override
+  bool shouldRepaint(CustomPainter oldDelegate) {
+    return true;
+  }
+}
+
+class SnakeState extends State<Snake> {
+  double offsetSize;
+  GameState state;
+  List<double> accelerometerValues;
```

Comment:
Maybe replace this by `AccelerometerEvent acceleration;` to avoid having to re-pack its values (line 70), and to enable use of `acceleration.x`,`acceleration.y` rather than `accelerometerValues[0]`,`accelerometerValues[1]` further below.